src/Aspire.Hosting/ApplicationModel/CustomResourceSnapshot.cs

```diff
@@ -77,6 +82,14 @@ public sealed record EnvironmentVariableSnapshot(string Name, string? Value, boo
 /// <param name="IsInternal">Determines if this url is internal.</param>
 public sealed record UrlSnapshot(string Name, string Url, bool IsInternal);
 
+/// <summary>
+/// A snapshot of a volume, mounted to a container.
+/// </summary>
+/// <param name="Source">The name of the volume. Can be <c>null</c> if the mount is an anonymous volume.</param>
+/// <param name="Target">The target of the mount.</param>
+/// <param name="IsReadOnly">Whether the volume mount is read-only or not.</param>
+public sealed record VolumeSnapshot(string? Source, string Target, bool IsReadOnly);
```

Comment:
I know the pattern here is to use records, but I have concerns about using records in public API. Let's say we add new fields in the future, how do we do that with a record?

I think a normal C# type with init properties would be easier to expand in the future.

Reply:
That's fair, but let's do that separately and keep the code consistent here.

Reply:
Agreed